src/GitHub.Exports/Primitives/UriString.cs

```diff
@@ -131,7 +134,9 @@ bool ParseScpSyntax(string scpString)
         /// <returns></returns>
         public Uri ToRepositoryUrl()
         {
-            if (url != null && IsFileUri) return url;
+            // we only want to process urls that represent network resources
+            if (!IsValidUri && !IsScpUri) return url;
```

Comment:
I guess my point is does this change simply push the nullref exception further up the stack?